include/deal.II/grid/reference_cell.h

```diff
@@ -3606,7 +3634,81 @@ ReferenceCell::face_to_cell_line_orientation(
     return numbers::default_geometric_orientation;
 }
 
-
+inline unsigned int
+ReferenceCell::opposite_face_index(const unsigned int face_no) const
+{
+  switch (this->kind)
+    {
+      case ReferenceCells::Vertex:
+        DEAL_II_NOT_IMPLEMENTED();
+      case ReferenceCells::Line:
+        {
+          AssertIndexRange(face_no, 2);
+          return (face_no + 1) % 2;
+        }
+      case ReferenceCells::Quadrilateral:
+        {
+          AssertIndexRange(face_no, GeometryInfo<2>::faces_per_cell);
```

Comment:
If you feel like it: All of these assertions can also just be moved to the top of the function, where you'd then write
```
  AssertIndexRange (face_no, n_faces());
```

I'll leave it to you whether you want to do this (or, if someone beats you to it by merging the PR, whether you want to do it as a follow-up patch).